### PR TITLE
added sign_message_hash_braavos function

### PR DESCRIPTION
**Pull request by tonypony220**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Feature

Resolves: #1077 
added  function sign deloy tx for braavos

-
-
-

## Does this introduce a breaking change?
No

### CODE REVIEW

File: crates/pallets/starknet/src/tests/deploy_account_tx.rs [hunk 1]
```diff
@@ -262,14 +262,12 @@ fn given_contract_run_deploy_account_braavos_tx_works() {
...
-        // Braavos has a complicated signature mecanism, they add stuffs around the tx_hash and then sign
-        // the whole thing. This hardcoded value is the expected "thing" that bravos expect you to
-        // sign for this transaction. Roll with it for now
-        let value_to_sign =
-            Felt252Wrapper::from_hex_be("0x06a8bb3d81c2ad23db93f01f72f987feac5210a95bc530eabb6abfaa5a769944").unwrap();
-        let mut signatures = sign_message_hash(value_to_sign);
-        signatures.extend_from_slice(&[Felt252Wrapper::ZERO; 8]);
-        deploy_tx.signature = signatures;
+        let tx_hash = deploy_tx.compute_hash::<<MockRuntime as Config>::SystemHash>(Starknet::chain_id(), false);
+        deploy_tx.signature = sign_message_hash_braavos::<<MockRuntime as Config>::SystemHash>(
```
Comment (EvolveArt) [inline crates/pallets/starknet/src/tests/deploy_account_tx.rs:266]: 2 small details:
1. We should probably not use the `SystemHash` here but directly the right hash for this braavos contract.
2. I prefer that we move the signature computation above the `deploy_tx` declaration with something like `let signature = sign...` then just fill it in the `deploy_tx` struct.

File: crates/pallets/starknet/src/tests/deploy_account_tx.rs [hunk 2]
```diff
@@ -262,14 +262,12 @@ fn given_contract_run_deploy_account_braavos_tx_works() {
...
+            tx_hash,
+            Felt252Wrapper::ZERO,
+            &[Felt252Wrapper::ZERO; 7],
```
Comment (EvolveArt) [inline crates/pallets/starknet/src/tests/deploy_account_tx.rs:269]: could we add a test with real values if possible ?

**Review by EvolveArt — CHANGES_REQUESTED** [review]
will merge once we have coverage back in CI

File: crates/pallets/starknet/src/tests/deploy_account_tx.rs (hunk 1 above)
Comment (tonypony220) [reply crates/pallets/starknet/src/tests/deploy_account_tx.rs]: Could you please clarify your second suggestion? It seems like signing a hash before obtaining the hash itself might not be possible.

File: crates/pallets/starknet/src/tests/deploy_account_tx.rs (hunk 2 above)
Comment (tonypony220) [reply crates/pallets/starknet/src/tests/deploy_account_tx.rs]: @EvolveArt Added hardcoded real signature in separate test.
I'm not sure if it's easy to add the secp256r1 curve, so we may do it in another issue if needed.

**Review by tdelabro — APPROVED** [review]

**Review by EvolveArt — APPROVED** [review]
lgtm, would be great to add more rustdoc/links to understand better the implementation

**Pull request merged**